Assets/HotmaxEditorVR/objectSelect.cs

```diff
@@ -5,174 +5,158 @@
-
-        hand1.gameObject.GetComponent<SteamVR_TrackedController>().controllerIndex = hand1ControllerIndex;
-        hand2.gameObject.GetComponent<SteamVR_TrackedController>().controllerIndex = hand2ControllerIndex;
+        _stateManagerMutatorRef.SET_SELECTED_OBJECT_IS_ACTIVE(false);
 
-        // convert to uint to an int 
-        hand1.gameObject.GetComponent<SteamVR_TrackedObject>().SetDeviceIndex((int)hand1ControllerIndex);
-        hand2.gameObject.GetComponent<SteamVR_TrackedObject>().SetDeviceIndex((int)hand2ControllerIndex);
+        //TODO: Is this the correct position for this?
+        _stateManagerMutatorRef.SET_ROTATION_GIZMO_IS_SELECTED(false);
     }
 
     void Update()
     {
-        ShootLaserFromTargetPosition(hand2.gameObject.transform.position, hand2.gameObject.transform.forward, laserMaxLength);
+        ShootLaserFromTargetPosition(inputManager.hand2.gameObject.transform.position, inputManager.hand2.gameObject.transform.forward, laserMaxLength);
     }
 
     void ShootLaserFromTargetPosition(Vector3 targetPosition, Vector3 direction, float length)
     {
         Ray ray = new Ray(targetPosition, direction);
         RaycastHit raycastHit;
-        endPosition = targetPosition + (length * direction);
+        laserEndPosition = targetPosition + (length * direction);
 
         if (Physics.Raycast(ray, out raycastHit, length))
         {
-            endPosition = raycastHit.point;
+            laserEndPosition = raycastHit.point;
         }
 
         laserLineRenderer.SetPosition(0, targetPosition);
-        laserLineRenderer.SetPosition(1, endPosition);
+        laserLineRenderer.SetPosition(1, laserEndPosition);
     }
 
 
     void select(Vector3 targetPosition, Vector3 direction)
     {
-        //TODO: write this so the event listener is inside this method so I can check in the update as well
         RaycastHit hit;
         Ray ray = new Ray(targetPosition, direction);
         if (Physics.Raycast(ray, out hit))
         {
-            //check to see if the raycast is hitting a game object
+
+            //validate that raycast is hitting something selectable
             if (hit.collider != null && !hit.collider.name.Contains("structure"))
             {
-                removeDecorators();
-                addDecorations(hit);           
-            }
 
-            if (hit.collider.name.Contains("spawnSlot"))
-            {
-               
-                if (hit.collider.name.Contains("(1)"))
-                {
-                    GameObject spawn = Instantiate(Resources.Load("(1)", typeof(GameObject))) as GameObject;
-                    spawn.transform.position = hit.collider.transform.position;
-                    spawn.transform.rotation = hit.collider.transform.rotation;
-                }
-                if (hit.collider.name.Contains("(2)"))
+                // selected object selection
+                if (hit.collider.gameObject == _selectedObject)
```

Comment:
This whole raycast check system needs to be smarter. 
The best way right now is create an Interface for raycastable stuff, extend anything that is raycastable with that interface, and then do a single check for typeOf raycastable -- and those classes should at that point handle a raycast hit however makes sense to them.
We'll go over this tomorrow, we'll need to study a little bit. I can see this getting out of hand if we don't tackle it now.

Reply:
https://unity3d.com/learn/tutorials/topics/scripting/interfaces